ideascube/mediacenter/views.py

```diff
@@ -47,9 +50,17 @@ def _set_available_kinds(self, context):
             (kind, label) for kind, label in Document.KIND_CHOICES
             if kind in available_kinds]
 
+    def _set_available_package_ids(self, context):
+        available_package_ids = self._search_for_attr_from_context('package_id', context)
+        catalog = catalog_mod.Catalog()
+
+        context['available_package_ids'] = {
+            package.id:package.name for package in catalog.list_installed(available_package_ids)}
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         self._set_available_kinds(context)
+        self._set_available_package_ids(context)
         self._set_available_langs(context)
         self._set_available_tags(context)
```

Comment:
Sure it is, call `self.get_context_data()`.

Store that in a cached property to avoid calling it multiple times.

But it probably would be more interesting to look back at that `_search_for_attr_from_context` method.

Anyway, like I said this isn't something for this PR, I merely wanted to mention it. (and I'm glad I did, since you apparently didn't know about it :smiley: )